### PR TITLE
fix bugs of insert and subquery

### PR DESCRIPTION
**Pull request by hanfei19910905**
@coocood @zimulala @shenli  PTAL
say a = all (select * from t) if t is a empty table, the result is true , and no matter what the operator is .
If the condition is any, the result is always false.
If a = any(subq)'s result is not true, and the rows in subq contains a null, the result is null, Or the result is false.
if a = all(subq)'s result is false, and the first condition that is not the true is NULL, the result is NULL. Otherwise it's false.

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -1583,6 +1589,18 @@ func (s *testSuite) TestNewSubquery(c *C) {
 	result.Check(testkit.Rows("1", "2", "3"))
 	result = tk.MustQuery("select t.c from t where (t.c, t.d) not in (select * from t)")
 	result.Check(testkit.Rows())
+	// = all empty set is true
+	result = tk.MustQuery("select t.c from t where (t.c, t.d) = all (select * from t where d > 1000)")
```
Comment (coocood) [inline executor/executor_test.go]: This statement should return "operands should has 1 column" error.

File: executor/new_executor.go [hunk 2]
```diff
@@ -1073,18 +1073,30 @@ type conditionChecker struct {
...
 	ctx     context.Context
 	all     bool
-	matched bool
+	data    *types.Datum
 }
 
-func (c *conditionChecker) Exec(row *Row) (*Row, error) {
-	var err error
-	c.matched, err = expression.EvalBool(c.cond, row.Data, c.ctx)
+func (c *conditionChecker) Check(row *Row) (*Row, error) {
```
Comment (coocood) [inline executor/new_executor.go]: `Check` indicates it is will not modify the input argument.

File: executor/new_executor.go [hunk 3]
```diff
@@ -1069,25 +1069,31 @@ type ApplyExec struct {
...
+	all         bool
+	dataHasNull bool
 }
 
-func (c *conditionChecker) Exec(row *Row) (*Row, error) {
-	var err error
-	c.matched, err = expression.EvalBool(c.cond, row.Data, c.ctx)
+// Check checks if the input row can determine the final result.
+// If so, it return the eval result for first result, and return true for second result.
+func (c *conditionChecker) Check(rowData []types.Datum) (*types.Datum, bool, error) {
```
Comment (coocood) [inline executor/new_executor.go]: We don't need to return Datum pointer.

File: expression/expression.go [hunk 4]
```diff
@@ -69,8 +69,8 @@ type Column struct {
 	// Position means the position of this column that appears in the select fields.
 	// e.g. SELECT name as id , 1 - id as id , 1 + name as id, name as id from src having id = 1;
 	// There are four ids in the same schema, so you can't identify the column through the FromID and ColName.
-	Position int
-	IsAgg    bool
+	Position    int
+	IsAggOrSubq bool
```
Comment (coocood) [inline expression/expression.go]: Add comments about this field.

**Comment by coocood** [comment]
LGTM

File: executor/new_executor.go [hunk 5]
```diff
@@ -1134,17 +1141,27 @@ func (e *ApplyExec) Next() (*Row, error) {
 			return srcRow, nil
 		}
 		if innerRow == nil {
-			srcRow.Data = append(srcRow.Data, types.NewDatum(e.checker.matched))
+			var d types.Datum
```
Comment (shenli) [inline executor/new_executor.go]: Add comment here.

**Comment by shenli** [comment]
LGTM

**Pull request merged**